hack/manifest-gen/assets/charts/eck/templates/_helpers.tpl

```diff
@@ -1,9 +1,74 @@
-{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "eck.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "eck.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
```

Comment:
Installation would fail in that case:

```
helm install elastic-operator ./hack/manifest-gen/assets/charts/eck -n elastic-system --create-namespace --set=fullnameOverride="my elastic operator" --set=image.repository=docker.elastic.co/eck-dev/eck-operator-cell --set=image.tag=1.3.0-SNAPSHOT-41932662 
Error: ServiceAccount "my elastic operator" is invalid: metadata.name: Invalid value: "my elastic operator": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

I haven't come across Helm charts that do validation of names and such. In the end, everything gets turned into Kubernetes resources anyway -- so if Kubernetes lets something through, it's probably valid.

I did introduce a few checks to catch impossible or non-sensical configurations. They are not exhaustive by any means because I am biased by insider knowledge and have some blind spots. We'll probably have to add more of those checks when we start seeing bug reports from users in the field.